app/services/duplicate_canonical_profile_resolver.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# Class that deduplicates canonical profiles with identical benchmark/ref IDs
+class DuplicateCanonicalProfileResolver
+  class << self
+    def run!
+      duplicates.each do |duplicate|
+        profiles = Profile.canonical.includes(:child_profiles).where(
+          duplicate.slice(:ref_id, :benchmark_id)
+        ).order(created_at: :asc).offset(1)
+
+        return duplicate_error_msg(duplicate) unless no_child_profiles? profiles
+
+        profiles.destroy_all
+        duplicate_success_msg(duplicate)
+      end
+    end
+
+    private
+
+    def duplicates
+      Profile.canonical.select(
+        :ref_id,
+        :benchmark_id,
+        :policy_id,
+        'COUNT(id)'
+      ).group(
+        :ref_id,
+        :benchmark_id,
+        :policy_id
+      ).having('count(id) > 1')
+    end
+
+    def no_child_profiles?(profiles)
+      profiles.all? { |p| p.child_profiles.count.zero? }
+    end
+
+    def duplicate_success_msg(duplicate)
+      Rails.logger.info(%(
+        Deleted #{duplicate.count} duplicates of policy with
+        ref_id=#{duplicate.ref_id} and benchmark_id=#{duplicate.benchmark_id}.
+      ))
+    end
```

Comment:
Shouldn't this be `s/duplicate/profiles/`? `duplicate.count` calls `count` on a profile vs. an ActiveRecord::Relation (i.e. `profiles`)

Reply:
Oh! gotcha, I see now :)